sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfigurationprovider/_azureappconfigurationprovider.py

```diff
@@ -0,0 +1,218 @@
+     KeyVaultSecretIdentifier
+)
+from azure.core.exceptions import (
+    HttpResponseError,
+    ResourceNotFoundError
+)
+from ._settingselector import SettingSelector
+from ._constants import (
+    KEY_VAULT_REFERENCE_CONTENT_TYPE
+)
+from ._user_agent import USER_AGENT
+
+
+class AzureAppConfigurationProvider:
+    """
+    Provides a dictionary-like interface to Azure App Configuration settings. Enables loading of sets of configuration
+    settings from Azure App Configuration into a Python application. Enables trimming of prefixes from configuration
+    keys. Enables resolution of Key Vault references in configuration settings.
+    """
+
+    def __init__(self):
+        # type: () -> None
+        self._dict = {}
+        self._trim_prefixes = []
+        self._client = None
+
+    @classmethod
+    def load(cls, connection_string=None, endpoint=None, credential=None, **kwargs):
+        """
+        Loads configuration settings from Azure App Configuration into a Python application.
+        :param connection_string: Connection string (one of connection_string or endpoint and credential must be set)
+        :type connection_string: str
+        :param endpoint: Endpoint (one of connection_string or endpoint and credential must be set)
+        :type endpoint: str
+        :param credential: Credential (one of connection_string or endpoint and credential must be set)
+        :type credential: Union[AppConfigConnectionStringCredential, TokenCredential]
+        :keyword selectors: List of setting selectors to filter configuration settings
+        :type selectors: list[~azure.appconfigurationprovider.SettingSelector]
+        :keyword trim_prefixes: List of prefixes to trim from configuration keys
+        :type trim_prefixes: list[str]
+        :keyword key_vault_options: Options for resolving Key Vault references
+        :type key_vault_options: ~azure.appconfigurationprovider.KeyVaultOptions
+        """
+        provider = AzureAppConfigurationProvider()
+
+        key_vault_options = kwargs.pop("key_vault_options", None)
+
+        provider.__buildprovider(connection_string, endpoint,
+                                 credential, key_vault_options)
+
+        selects = kwargs.pop("selects", {SettingSelector("*", "\0")})
+
+        provider._trim_prefixes = sorted(kwargs.pop(
+            "trimmed_key_prefixes", []), key=len, reverse=True)
+
+        provider._dict = {}
+
+        secret_clients = key_vault_options.secret_clients if key_vault_options else {}
+
+        for select in selects:
+            try:
+                configurations = provider._client.list_configuration_settings(
+                    key_filter=select.key_filter, label_filter=select.label_filter)
+            except HttpResponseError as e:
+                raise e
+            for config in configurations:
+                if config.content_type == KEY_VAULT_REFERENCE_CONTENT_TYPE:
+                    provider.__resolve_keyvault_references(
```

Comment:
Can we return the secret value from `__resolve_keyvault_references` instead of modifying the state of `self._dict` there? THe method that resolves Key Vault refs should not know about trimming key prefixes.
Also, should it be called `__resolve_keyvault_reference` since we only resolve one reference at a time?